core/include/traccc/finding/finding_algorithm.ipp

```diff
@@ -19,29 +19,48 @@ namespace traccc {
 template <typename stepper_t, typename navigator_t>
 track_candidate_container_types::host
 finding_algorithm<stepper_t, navigator_t>::operator()(
-    const detector_type& det,
-    const measurement_container_types::host& measurements,
+    const detector_type& det, measurement_collection_types::host&& measurements,
     const bound_track_parameters_collection_types::host& seeds) const {
 
     track_candidate_container_types::host output_candidates;
 
-    /**********************
-     * Create module map
-     **********************/
-
-    std::vector<std::pair<geometry_id, unsigned int>> module_map;
-    const int n_modules = measurements.size();
+    /*****************************************************************
+     * Measurement Operations
+     *****************************************************************/
+
+    // Sort the measurements w.r.t geometry barcode
+    std::sort(measurements.begin(), measurements.end(),
+              measurement_sort_comp());
```

Comment:
Okay, so you move the measurement container to avoid making a copy for the sorting... :thinking:

I'm not a fan of this. But I see how one could argue that this could provide an optimisation in the cases where the measurement collection is not needed after the track finding anymore.

Is that the case by the way? Does track fitting not need the measurements anymore? Because if it does, then we're only making the user code around the algorithms more complicated than they need to be. Shifting the copy-making to the user of this algorithm, instead of the algorithm making the copy itself. :thinking:

Reply:
I don't think we need measurement collection after track finding.
In that case, do you think it is better to make a copy of the measurement collection for sorting?

Reply:
Just for update: I changed to use const reference and make a copy for sorted measurements